app/lib/shared/platform.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:pana/pana.dart';
+
+abstract class KnownPlatforms {
+  static const String flutter = PlatformNames.flutter;
+  static const String server = PlatformNames.server;
+  static const String web = PlatformNames.web;
+  static const List<String> everywhere = const [flutter, server, web];
+
+  static bool isPredicate(String platform) => everywhere.contains(platform);
+}
+
+List<String> indexDartPlatform(DartPlatform platform) {
+  if (platform == null) {
+    return null;
+  }
+  if (platform.worksEverywhere) {
+    return KnownPlatforms.everywhere;
+  }
+  if (platform.restrictedTo == null || platform.restrictedTo.isEmpty) {
+    return null;
+  }
+  return new List.from(platform.restrictedTo);
+}
+
+class PlatformPredicate {
+  final List<String> required;
+  final List<String> prohibited;
```

Comment:
Why do we have a `DartPlatform` class if we then represent a `platform` mostly as plain `String`s?